docs/model-dev-guide/apis-howto/deepspeed/autotuning.rst

```diff
@@ -0,0 +1,303 @@
+experiments is as easy as replacing the usual experiment-launching command
+
+.. code::
+
+   det experiment create deepspeed.yaml .
+
+with
+
+.. code::
+
+   python3 -m determined.pytorch.dsat asha deepspeed.yaml .
+
+The above uses Determined AI's DeepSpeed Autotune with the ``asha`` algorithm, one of three
+available search methods:
+
+-  ``asha``: Adaptively searches over randomly selected DeepSpeed configurations, allocating more
+   compute resources to well-performing configurations. See :ref:`this introduction to ASHA
+   <topic-guides_hp-tuning-det_adaptive-asha>` for more details.
+
+-  ``binary``: Performs a simple binary search over the batch size for randomly-generated DS
+   configurations.
+
+-  ``random``: Conducts a search over random DeepSpeed configurations with an aggressive
+   early-stopping criteria based on domain-knowledge of DeepSpeed and the search history.
+
+DeepSpeed Autotune is built on top of Custom Searcher (see :ref:`topic-guides_hp-tuning-det_custom`)
+which starts up two separate experiments:
+
+-  ``single`` Search Runner Experiment: This experiment coordinates and schedules the trials that
+   run the model code.
+-  ``custom`` Experiment: This experiment contains the trials referenced above whose results are
+   reported back to the search runner.
+
+Initially, a one-step profiling trial is created to gather information regarding the model and
+computational resources. The search runner experiment takes this initial profiling information and
+creates a series of trials to search for the DS settings which optimize ``FLOPS_per_gpu``,
+``throughput`` (samples/second), or latency timing information. The results of all such trials can
+be viewed in the ``custom`` experiment above. The search is informed both by the initial profiling
+trial and the results of each subsequent trial, all of whose results are fed back to the search
+runner.
+
+*******************
+ User Code Changes
+*******************
+
+To use ``dsat`` with :class:`~determined.pytorch.deepspeed.DeepSpeedTrial`, Core API, and
+HuggingFace Trainer, specific changes must be made to your user code. In the following sections, we
+will describe these changes in detail for each use case.
+
+.. _using_deepspeed_trial:
+
+DeepSpeedTrial
+==============
+
+Determined's DeepSpeed Autotune works by inserting DS configuration options into the
+``overwrite_deepspeed_args`` field of the ``hyperparameters`` dictionary which is seen by each
+trial. To take advantage of ``dsat``, you simply need to incorporate these overwrite values into
+your original configuration.
+
+.. note::
+
+   For more information about ``DeepSpeedTrial`` see :ref:`deepspeed-api`
+
+To facilitate this process, you must add a ``deepspeed_config`` field under the ``hyperparameters``
+section of your experiment. This field specifies the relative path to the DS ``json`` configuration
+file (written following the `DS documentation here <https://www.deepspeed.ai/docs/config-json/>`_)
```

Comment:
We should probably notate this as the "DeepSpeed documentation" rather than as "DS documentation"